src/commands/crashes/upload-mappings.ts

```diff
@@ -0,0 +1,88 @@
+import { AppCenterClient, models } from "../../util/apis";
+import { AppCommand, CommandResult } from "../../util/commandline";
+import { ErrorCodes, failure, success } from "../../util/commandline";
+import { hasArg, help, longName, shortName } from "../../util/commandline";
+import { out } from "../../util/interaction";
+import { inspect } from "util";
+import { DefaultApp } from "../../util/profile";
+import UploadSymbolsHelper from "./lib/symbols-uploading-helper";
+import { SymbolType } from "./lib/symbols-uploading-helper";
+
+import * as Fs from "fs";
+import * as Path from "path";
+
+import * as _ from "lodash";
+
+const debug = require("debug")("appcenter-cli:commands:apps:crashes:upload-mappings");
+
+@help("Upload the Android mappings for the application")
+export default class UploadMappings extends AppCommand {
+  @help("Path to an Android mapping.txt file.")
+  @shortName("m")
+  @longName("mapping")
+  @hasArg
+  public mappingPath: string;
+
+  @help("The version name to associate with the mappings.")
+  @shortName("n")
+  @longName("version-name")
+  @hasArg
+  public versionName: string;
+
+  @help("The version code to associate with the mappings.")
+  @shortName("c")
+  @longName("version-code")
+  @hasArg
+  public versionCode: string;
+
+  public async run(client: AppCenterClient): Promise<CommandResult> {
+    const app: DefaultApp = this.app;
+
+    this.validateParameters();
+
+    this.getStatsForFsPath(this.mappingPath);
+    if (this.getLowerCasedFileExtension(this.mappingPath) !== ".txt") {
+      throw failure(ErrorCodes.InvalidParameter, `path ${this.mappingPath} does not point to valid mapping.txt file`);
+    }
+    const uploadRequest: models.SymbolUploadBeginRequest = {
+      symbolType: SymbolType.AndroidProGuard,
+      fileName: "mapping.txt",
+      version: this.versionName,
+      build: this.versionCode
+    };
+
+    // upload mappings
+    await out.progress("Uploading mappings...", new UploadSymbolsHelper(client, app, debug).uploadSymbolsArtifact(this.mappingPath, uploadRequest));
+
+    return success();
+  }
+
+  private getStatsForFsPath(filePath: string): Fs.Stats | null {
+    // take fs entry stats (and check it's existence BTW)
+    try {
+      debug(`Getting FS statistics for ${filePath}`);
+      return Fs.statSync(filePath);
+    } catch (error) {
+      if (error.code === "ENOENT") {
+        // path points to non-existing file system entry
+        throw failure(ErrorCodes.InvalidParameter, `path ${filePath} points to non-existent item`);
+      } else {
+        // other errors
+        debug(`Failed to get statistics for file system entry ${filePath} - ${inspect(error)}`);
+        throw failure(ErrorCodes.Exception, `failed to get statistics for file system entry ${filePath}`);
+      }
+    }
+  }
+
+  private getLowerCasedFileExtension(filePath: string): string {
```

Comment:
This is used for finding the filename extension in line 44.

Reply:
Do we need a 1-line helper function that isn't reused elsewhere? I think we could just directly call `Path.extname(this.mappingPath).toLowerCase()` in Line 44, no?

Reply:
Fair enough - I didn't read your message as "are we reusing this" but just "are we using this" - I don't have strong feelings here, it is another kind where we took the same approach as `upload-symbols` but since it's private, we can just inline.